### PR TITLE
upgrade-envoy-v2

### PR DESCRIPTION
**Pull request by danpilch**
- **WIP moving to v3 discovery API**
- **Upgrade Envoy gRPC API to V3**
- **Fix tests**
- **go mod tidy**
- **changes to docker copy path**
- **HttpFilter ConfigType router config required for v3 > v1.20.x envoy**
- **bump some go packages for envoy**
- **haproxy deprecated ':' valid in config**

testing against envoy `v1.29.7`

### CODE REVIEW

File: envoy/adapter/adapter.go [hunk 1]
```diff
@@ -213,18 +224,22 @@ func EnvoyResourcesFromState(state *catalog.ServicesState, bindIP string,
...
+		fallthrough
 	case "http":
 		managerName = wellknown.HTTPConnectionManager
 
+		routerConfig, _ := anypb.New(&router.Router{})
 		manager = &hcm.HttpConnectionManager{
 			StatPrefix: "ingress_http",
 			HttpFilters: []*hcm.HttpFilter{{
 				Name: wellknown.Router,
+				ConfigType: &hcm.HttpFilter_TypedConfig{TypedConfig: routerConfig},
```
Comment (danpilch) [inline envoy/adapter/adapter.go:239]: required in envoy > v1.20.x else envoy spews errors

File: views/haproxy.cfg [hunk 2]
```diff
@@ -27,7 +27,7 @@ defaults
 frontend stats_proxy
 	mode http
 	bind 0.0.0.0:3212
-	http-response add-header Access-Control-Allow-Origin: *
+	http-response add-header Access-Control-Allow-Origin "*"
```
Comment (danpilch) [inline views/haproxy.cfg:30]: newer haproxy fails to start with `:`

**Comment by danpilch** [comment]
@mihaitodor I created another PR based on your branch with my changes on top

**Comment by danpilch** [comment]
testing against envoy 1.29.7

**Review by mihaitodor — APPROVED** [review]
Nice one @danpilch, thanks for taking the time to go through my code! Happy to see this finally merged :) The extra commits you added look good to me. The Envoy API makes some of this stuff a royal pain. Feel free to :shipit: if you're happy with it.

Note that the ingress config generator to will likely need some work as well if you're still using Envoy on the ingress side. I didn't get around to seeing what may need to change there.

File: envoy/adapter/adapter.go [hunk 3]
```diff
@@ -257,44 +279,6 @@ func connectionManagerForService(svc *service.Service, envoyServiceName string)
 				Cluster: envoyServiceName,
 			},
 		}
-	case "ws":
```
Comment (relistan) [inline envoy/adapter/adapter.go:260]: You removed websocket support it looks like?

Comment (mihaitodor) [reply envoy/adapter/adapter.go:260]: Nope, IIRC I just deduplicated the code. See [here](https://github.com/NinesStack/sidecar/blob/6a309151b1de50830b0a985380ad77fc1ba70c76/envoy/adapter/adapter.go#L229-L230) and [here](https://github.com/NinesStack/sidecar/blob/6a309151b1de50830b0a985380ad77fc1ba70c76/envoy/adapter/adapter.go#L266-L272).

**Pull request merged**